plugins/orchestrator/src/components/ExecuteWorkflowPage/ExecuteWorkflowPage.tsx

```diff
@@ -66,42 +76,61 @@ export const ExecuteWorkflowPage = (props: ExecuteWorkflowPageProps) => {
       });
   }, [orchestratorApi, workflowId]);
 
-  const onExecute = useCallback(async () => {
-    const parameters: Record<string, JsonValue> = {};
-    if (schemaResponse?.schema && formState) {
-      for (const key in formState) {
-        if (formState.hasOwnProperty(key)) {
-          const property = formState[key];
-          Object.assign(parameters, { [key]: property });
+  const onExecute = useCallback(
+    async (e: FormEvent) => {
+      e.preventDefault();
+
+      if (!schemaResponse) {
+        return;
+      }
+
+      let parameters: Record<string, JsonValue> = {};
+      if (schemaResponse.dataInputSchema?.mainSchema) {
+        if (!formRef.current?.validateForm()) {
+          return;
+        }
+        // Flatten parameters are only necessary for this case where we show one form for all ref schemas (steps).
+        parameters = flattenParametersFromFormState(formState);
+      } else {
+        if (!jsonTextAreaRef.current?.validate()) {
+          return;
         }
+        parameters = JSON.parse(jsonTextAreaRef.current.getContent());
       }
-    }
 
-    setLoading(true);
-    const workflowCategory = getWorkflowCategory(
-      schemaResponse?.workflowItem.definition,
-    );
-    if (workflowCategory === WorkflowCategory.ASSESSMENT) {
-      Object.assign(parameters, { businessKey: crypto.randomUUID() });
-    } else {
-      Object.assign(parameters, { businessKey: businessKey ?? '' });
-    }
-    const response = await orchestratorApi.executeWorkflow({
+      const workflowCategory = getWorkflowCategory(
+        schemaResponse?.workflowItem.definition,
+      );
+
+      if (businessKey) {
```

Comment:
perhaps the handling of buisnessKey should move to the backend?
it's an implementation details

Reply:
The business key can be provided through the [route](https://github.com/caponetto/backstage-plugins/blob/KOGITO-9778-setup/plugins/orchestrator/src/routes.ts#L59) so we have to pass it to BE anyway.

Reply:
IIUC there's also a [route](https://github.com/caponetto/backstage-plugins/blob/KOGITO-9778-setup/plugins/orchestrator/src/routes.ts#L43) that doesn't accept a buisness key
so maybe if there's no buisness key we don't need to pass this?

Reply:
The current behavior is to always pass a business key if the workflow is Assessment.
We'll have to check later if this behavior is correct.
Anyway, I just moved the code around, the logic is kept in this area.

Reply:
yes I understand it's the same but since I'm reviewing the changes I noticed this
I'm just wondering about:
parameters.buisnessKey = uid.v4()
If it's an assessment workflow without a  buisness key it doesn't make sense to me that the UI should generate it
And sinse IIUC correctly the buisness key isn't required in the route, this line can be removed

Reply:
I understand your concern, but let's check with @RichardW98 once he's back.

Reply:
It's not blocking for this PR, we can check it later